mirrord-agent/src/steal.rs

```diff
@@ -106,6 +105,47 @@ impl Drop for SafeIpTables {
     }
 }
 
+enum IPTableFormatter {
+    Normal,
+    Linkerd,
+}
+
+impl IPTableFormatter {
+    fn detect(ipt: &iptables::IPTables) -> Result<Self> {
+        let prerouting = ipt
+            .list(IPTABLES_TABLE_NAME, "OUTPUT")
+            .map_err(|e| AgentError::IPTablesError(e.to_string()))?;
+
+        if prerouting
+            .iter()
+            .any(|rule| rule.contains("-j PROXY_INIT_OUTPUT"))
+        {
+            Ok(IPTableFormatter::Linkerd)
+        } else {
+            Ok(IPTableFormatter::Normal)
+        }
+    }
+
+    fn entrypoint(&self) -> &str {
+        match self {
+            IPTableFormatter::Normal => "PREROUTING",
+            IPTableFormatter::Linkerd => "OUTPUT",
+        }
+    }
+
+    fn redirect_rule(&self, redirected_port: Port, target_port: Port) -> String {
+        let redirect_rule = format!(
+            "-m tcp -p tcp --dport {} -j REDIRECT --to-ports {}",
+            redirected_port, target_port
+        );
+
+        match self {
+            IPTableFormatter::Normal => redirect_rule,
+            IPTableFormatter::Linkerd => format!("-o lo {}", redirect_rule),
```

Comment:
I think somewhat better implementation would be a trait then func that returns the appropriate struct (by using `dyn Impl`?) but it's a bit a nitpick ofc.
also some UT here would be nice

Reply:
you can make a trait and accept stuff that `impl` it. maybe @meowjesty have better idea